core/src/main/scala/chisel3/properties/Object.scala

```diff
@@ -21,6 +25,15 @@ class DynamicObject private[chisel3] (val className: ClassType) extends HasId wi
 
   _parent.foreach(_.addId(this))
 
+  // Keep state for a reference to the Class from which the DynamicObject was created.
+  // This is used to update the Class ref to the DynamicObject ref, for Classes created via DynamicObject.apply.
+  private var _class: Option[Class] = None
+  protected[chisel3] def setSourceClass(cls: Class): Unit = {
+    require(!_class.isDefined, "Cannot set DynamicObject class multiple times")
+    _class = Some(cls)
+  }
+  protected[chisel3] def getSourceClass: Option[Class] = _class
```

Comment:
```suggestion
  private var _class: Class = null
  protected[chisel3] def setSourceClass(cls: Class): Unit = {
    require(_class != null, "Cannot set DynamicObject class multiple times")
    _class = cls
  }
  protected[chisel3] def getSourceClass: Option[Class] = Option(_class)
```
It's a micro-optimization but one that saves a fair amount of memory if we end up having a lot of `DyanmicObjects` for whatever reason. We've done this same optimization for similar fields in `Data`.